pkg/server/registry/apigroups/acorn/events/strategy.go

```diff
@@ -127,22 +137,33 @@ func (q query) filter(events ...apiv1.Event) []apiv1.Event {
 		tail = int(q.tail)
 	}
 
-	events = events[len(events)-tail:]
+	if q.details && q.prefix.all() {
+		// Query selects all remaining events and includes details
+		return events[len(events)-tail:]
+	}
 
-	if q.details {
-		return events
+	results := make([]apiv1.Event, 0, tail)
+	for _, event := range events {
+		if !q.prefix.matches(event) {
+			// Exclude from results
+			continue
+		}
+
+		if !q.details {
+			event.Details = nil
+		}
+
+		results = append(results, event)
```

Comment:
Looking for some guidance here.

I think the _correct_ way to do this would be by indexing at the storage layer and querying that index, but I'm not sure if that's something we've exposed in mink yet; especially for remote backed strategies. I scanned through a few times but didn't find anywhere else we're doing something like this.

Reply:
I think this current approach is fine. I'm also not aware of anything in mink that would let us do something like that.